src/resolve/metadataResolver.ts

```diff
@@ -369,8 +369,21 @@ const resolveType =
  */
 const parseAsContentMetadataXml =
   (registry: RegistryAccess) =>
-  (fsPath: string): boolean =>
-    Boolean(registry.getTypeBySuffix(extName(fsPath)));
+  (fsPath: string): boolean => {
+    const suffixType = registry.getTypeBySuffix(extName(fsPath));
+    if (!suffixType) return false;
+
+    const matchesSuffixType = fsPath.split(sep).includes(suffixType.directoryName);
+    if (matchesSuffixType) return matchesSuffixType;
```

Comment:
for an XML file, `suffixType` would return the `emailservicesfunction` entry:
https://github.com/forcedotcom/source-deploy-retrieve/blob/9d94840fa9523c2e9d4268f231c7f889ab8160b0/src/registry/metadataRegistry.json#L522